libimage/pull.go

```diff
@@ -71,6 +72,38 @@ func (r *Runtime) Pull(ctx context.Context, name string, pullPolicy config.PullP
 	if err != nil {
 		return nil, err
 	}
+	rwDirs := func() []string {
+		rw := []string{}
+		for _, d := range []string{
+			r.store.RunRoot(),
+			r.store.GraphRoot(),
+			"/tmp",
+			"/var/tmp",
+			defaultConfig.Engine.ImageCopyTmpDir,
+			defaultConfig.Engine.TmpDir,
+		} {
+			if _, err := os.Stat(d); err == nil {
+				rw = append(rw, d)
+			}
+		}
+		return rw
+	}
+
+	if err := landlock.V5.BestEffort().RestrictPaths(
```

Comment:
>  RestrictPaths restricts all goroutines

That seems entirely unsuitable for processes not doing exclusively pulls, notably CRI-O, but probably also some Podman uses.

Reply:
I have been trying to get a test program to work, and failing miserably.
[t.go](https://github.com/user-attachments/files/16689843/t.go.txt)